doc/rst/developer/bestPractices/ContributorInfo.rst

```diff
@@ -9,184 +9,274 @@ The git repository for the project is hosted on Github at
+  on other deadlines.
 
-Mail chapel-developers@ for review and merge
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+* See `Reviewer responsibilities`_ for details on what performing a review on
+  another contributor's code entails.
 
-* When you believe your pull request is ready for review, send it to a member
-  of the core Chapel team (or chapel-developers_ if there isn't an obvious
-  person to review it). Even the developers that have write access to the
-  Chapel repository need to have all non-trivial changes reviewed. Developers
-  who have been given write access can merge trivial changes (e.g. small bug
-  fixes, documentation changes) without review.
+.. _Work with your reviewers:
+
+Work with your reviewers
+++++++++++++++++++++++++
+
+* Iterate with the reviewer until you're both satisfied.  One should generally
+  try to do whatever their reviewer asks.  Sometimes, a reviewer will ask for
+  something really hard.  Try to make sure they understand the magnitude of the
+  request, and try to discuss if it's really necessary to do before merging.  If
+  you can't come to agreement, one of you should bring other developers
+  (individually or via chapel-developers_) into the conversation to get a
+  broader opinion (one of the jobs of the reviewer is to serve as a proxy for
+  other developers, or to bring those developers into the conversation if they
+  feel unqualified to do so).
+
+
+.. _Test changes:
+
+Test changes
+~~~~~~~~~~~~
 
-  .. note::
+If you did not have the resources to perform at least a full correctness test
+run, this is the point at which a reviewer would do so for you.  Contributors
+are expected to have verified any new tests work before asking a reviewer to do
+this.
 
-    Ideally, someone should volunteer to review your pull request within a day
-    or two. If this doesn't happen, feel free to make some noise. Ideally the
-    review should take place within a few days, though timing may vary
-    depending on other deadlines.
+Details on how to run tests may be found at `Test System`_, and details on the
+appropriate amount of testing before merging the final PR may be found at
+`Testing guidance`_.
 
-* Iterate with the reviewer until you're both satisfied. If you can't come to
-  agreement, one of you should bring other developers (individually or via
-  chapel-developers_) into the conversation to get a broader opinion (one of
-  the jobs of the reviewer is to serve as a proxy for other developers, or to
-  bring those developers into the conversation if they feel unqualified to do
-  so).
+.. _Getting ready to merge:
+
+Getting ready to merge
+~~~~~~~~~~~~~~~~~~~~~~
+
+Before the change can be merged, ensure:
+- all design changes have been discussed
+- the PR has been reviewed
+- the contributor agreement has been signed
+- the history in the PR does not contain anything not to be in the repo
+- relevant configurations pass testing
```

Comment:
I like this one as a bulleted list, but the formatting isn't working now. Also, could it use the word "checklist" in the  topic sentence?